integration/test-integration/src/test/java/org/wso2/choreo/connect/tests/testcases/withapim/soap/SOAPApisTestCase.java

```diff
@@ -0,0 +1,116 @@
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.choreo.connect.tests.testcases.withapim.soap;
+
+import com.github.dockerjava.zerodep.shaded.org.apache.hc.core5.http.HttpStatus;
+import org.testng.Assert;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Test;
+import org.wso2.choreo.connect.tests.apim.ApimBaseTest;
+import org.wso2.choreo.connect.tests.apim.ApimResourceProcessor;
+import org.wso2.choreo.connect.tests.apim.utils.StoreUtils;
+import org.wso2.choreo.connect.tests.common.model.API;
+import org.wso2.choreo.connect.tests.context.CCTestException;
+import org.wso2.choreo.connect.tests.util.TestConstant;
+import org.wso2.choreo.connect.tests.util.TokenUtil;
+import org.wso2.choreo.connect.tests.util.Utils;
+import org.wso2.choreo.connect.tests.util.HttpResponse;
+import org.wso2.choreo.connect.tests.util.HttpsClientRequest;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+public class SOAPApisTestCase extends ApimBaseTest {
+    private static final String API_NAME = "SoapAPI";
+    private static final String API_CONTEXT = "soap";
+    private static final String APPLICATION_NAME = "DefaultAPIApp";
+    private static final String API_VERSION = "1.0.0";
+    private static final String SOAP_ACTION = "http://mockbackend:2340/phoneverify/query/CheckPhoneNumber";
+    private final Map<String, String> requestHeaders_11 = new HashMap<>();
+    private final Map<String, String> requestHeaders_12 = new HashMap<>();
+
+    String internalKey;
+    String endpointURL11;
+    String endpointURL12;
+
+    @BeforeClass(alwaysRun = true, description = "Create access token and define endpoint URL")
+    void setEnvironment() throws Exception {
+        super.initWithSuperTenant();
+        // Get App ID and API ID
+        String applicationId = ApimResourceProcessor.applicationNameToId.get(APPLICATION_NAME);
+        String apiId = ApimResourceProcessor.apiNameToId.get(API_NAME);
+
+        String accessToken = StoreUtils.generateUserAccessToken(apimServiceURLHttps, applicationId,
+                user, storeRestClient);
+        requestHeaders_11.put(TestConstant.AUTHORIZATION_HEADER, "Bearer " + accessToken);
+        requestHeaders_11.put(TestConstant.CONTENT_TYPE_HEADER, TestConstant.CONTENT_TYPES.TEXT_XML);
+        requestHeaders_11.put(TestConstant.SOAP_ACTION_HEADER, SOAP_ACTION);
+        requestHeaders_12.put(TestConstant.AUTHORIZATION_HEADER, "Bearer " + accessToken);
+        requestHeaders_12.put(TestConstant.CONTENT_TYPE_HEADER, TestConstant.CONTENT_TYPES.SOAP_XML);
+        API api = new API();
+        api.setContext(API_CONTEXT + TestConstant.URL_SEPARATOR + "1.0.0");
+        api.setName(API_NAME);
+        api.setVersion("1.0.0");
+        api.setProvider("admin");
+
+        internalKey = TokenUtil.getJWT(api, null, "Unlimited", TestConstant.KEY_TYPE_PRODUCTION,
+                3600, null, true);
```

Comment:
```suggestion
```

Reply:
fixed in https://github.com/wso2/product-microgateway/pull/2972/commits/5c91048f9fa451e4b2f2d91dac704d3f6b755cd3